packager/media/formats/mp4/mp4_media_parser.h

```diff
@@ -36,6 +36,7 @@ class MP4MediaParser : public MediaParser {
   void Init(const InitCB& init_cb,
             const NewMediaSampleCB& new_media_sample_cb,
             const NewTextSampleCB& new_text_sample_cb,
+            const DASHEventMessageBoxCB& event_message_cb,
```

Comment:
The `emsg` boxes only occur in mp4 containers, so we can add the callback to this class only without having to change the `MediaParser::Init()` interface.  This allows the whole class hierarchy and other files remain untouched.

For instance, one way we can accomplish this is by adding a member function and then calling it in the demuxer (for the appropriate container type).
```
void MP4MediaParser::SetEventMessageBoxCB(const DASHEventMessageBoxCB& event_message_cb) {
  event_message_cb_ = event_message_cb;
}
```